src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20171002_cluster-init-bucket-column.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="leonardo" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet logicalFilePath="leonardo" author="hussein" id="cluster-init-bucket">
+        <addColumn tableName="CLUSTER">
+            <column name="initBucket" type="varchar(1024)"/>
+        </addColumn>
+    </changeSet>
+
+    <changeSet logicalFilePath="leonardo" author="hussein" id="longer-user-bucket">
+        <modifyDataType columnName="googleBucket" newDataType="varchar(1024)" tableName="CLUSTER"/>
```

Comment:
Since bucket names are max 63 characters, `varchar(1024)` might not be necessary. It would be if this were a bucket _path_ instead of a bucket name.

Reply:
but it might well be a path, right? i mean, there's nothing to stop the user passing gs://foo/bar to Leo. (or IMO there shouldn't be.)

Reply:
If we're getting a `Specified key was too long; max key length is 767 bytes` error on migration, should we just use the bucket name and not the path so that we can make the key length smaller?

Reply:
My guess is that the issue here is MySQL doesn't like long unique keys. I think users are gonna want to use a random bucket path as a folder rather than just a top-level bucket, so I'm gonna de-uniqify the column and see if it makes the problem go away.

Reply:
https://dev.mysql.com/doc/refman/5.6/en/innodb-restrictions.html#innodb-mximums-minimums

Reply:
+1 on de-uniquifying bucket columns